validators/btc_address.py

```diff
@@ -54,7 +52,8 @@ def btc_address(value: str, /):
     """
     if value and type(value) is str:
         return (
-            _segwit_pattern.match(value)
+            # segwit pattern
+            re.compile(r"^(bc|tc)[0-3][02-9ac-hj-np-z]{14,74}$").match(value)
```

Comment:
Why not compile this regex outside of this function? Now it needs to be compiled everytime this function is called. Also the users sometimes might want to re-use the regex somewhere else. Now it is not possible since it is not available in a variable.

Reply:
For the following reasons.

- I do not think the `validators` package aims to provide regular expressions from its API. Users always have access to the source, so they can copy regex whenever they want.
- If they want to perform validation with custom or modified regex, would they still use `validators` library? And if they do, I think it'll be neater to provide it as a function parameter.

On memory cost..

- Earlier when users do `import validators`, the regexps are compiled and stored in module level namespace variables, which claims some memory, _throughout the users' program's lifetime_.
- This memory footprint is a side-effect for just importing the module. In this case the it is quite negligible. But I have had experiences with other libraries like `loguru`  and `Scrapy` which adds this memory penalty.
- When it is inevitably necessary (like sharing same pattern across multiple modules), we can have such entities outside function definition. 

On computational cost..

- Yes this PR and #233 will add computational cost on every call. I suspect the number of calls may be generally infrequent, unless someone's trying validate a big file of email addresses, looping through each of one them.
- To address this I can use a separate function to compile regexp and cache it.

Reply:
Ok makes sense 👍